syntax = "proto3";

package keto.proto;

/**
 * This is an example of a protobuf message.
 */
message SessionKeyRequest {
    bytes session_hash = 1;
}

message SessionKeyResponse {
    bytes session_hash = 1;
    bytes session_key = 2;
}

message SessionKeyExpireRequest {
    bytes session_hash = 1;
}

message SessionKeyExpireResponse {
    bytes session_hash = 1;
    bool  success = 2;
}

message NetworkKey {
    bytes key_hash = 1;
    bytes network_key = 2;
    bool active = 3;
}

message NetworkKeys {
    uint32 slot = 1;
    uint64 timestamp = 2;
    repeated NetworkKey network_keys = 3;
}

message NetworkKeysWrapper {
    bytes network_keys_bytes = 1;
}

message MasterInfo {
    bool is_master = 1;
}


message EncryptRequest {
    uint32 version = 1;
    bytes asn1_message = 2;
}

message EncryptResponse {
    uint32 version = 1;
    bytes asn1_message = 2;
    bytes network_bytes_message = 3;
}

message EncryptNetworkRequest {
    uint32 version = 1;
    bytes bytes_message = 2;
}

message EncryptNetworkResponse {
    uint32 version = 1;
    bytes bytes_message = 2;
}


message PasswordRequest {
    bytes identifier = 1;
}

message PasswordResponse {
    bytes identifier = 1;
    bytes password = 2;
}
